src/esthri/src/main.rs

```diff
@@ -385,16 +384,10 @@ async fn dispatch_esthri_cli(cmd: EsthriCommand, s3: &S3Client) -> Result<()> {
             ..
         } => {
             setup_upload_termination_handler();
-            #[cfg(feature = "compression")]
-            {
-                if matches!(cmd, Put { compress: true, .. }) {
-                    upload_compressed(s3, bucket, key, file).await?;
-                } else {
-                    upload(s3, bucket, key, file).await?;
-                }
-            }
-            #[cfg(not(feature = "compression"))]
-            {
+
+            if matches!(cmd, Put { compress: true, .. }) {
+                upload_compressed(s3, bucket, key, file).await?;
+            } else {
                 upload(s3, bucket, key, file).await?;
             }
         }
```

Comment:
One last tiny nit but I think you can skip these `matches!` and it's a bit clearer

```rust
match cmd {
        Put {
            ref bucket,
            ref key,
            ref file,
            compress,
        } => {
            setup_upload_termination_handler();
            if compress {
                upload_compressed(s3, bucket, key, file).await?;
            } else {
                upload(s3, bucket, key, file).await?;
            }
        },
        ...
}
```